### PR TITLE
feat(client): language dropdown in the side menu is now a drop down

### LINKED ISSUE #42951: New languages in and long strings in nav menu
**Affected page**
All pages on /learn (mainly Portuguese shows up the issue)

**To Reproduce**

1. Go to https://www.freecodecamp.org/portuguese/
2. Click on 'Menu'
4. The string for "Thanks for donating" is too large to fit.

**Expected behavior**

1. Need to figure out a scalable way of adding languages to the list.
2. The phrases on the menu should wrap.

**Screenshots**

image

### PR DESCRIPTION
**Pull request by shangyilim**
As suggested in #42951  I have made the following changes:
1. Make text wrap in the navigation menu items.
2. Added language dropdown menu in place of the language selection.
3. Prevent language dropdown menu from closing when selecting a language until after a language is selected.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #42951

Screenshots:
1. Text wrap
image
2. Language Dropdown
image
image

### CODE REVIEW

**Review by raisedadead — DISMISSED** [review]
Thanks for this PR. This looks great.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Hi @shangyilim this looks neat, nice job!  I did find a couple of issues that need addressing, though:

File: client/src/components/Header/components/nav-links.tsx [hunk 1]
```diff
@@ -167,32 +186,19 @@ export class NavLinks extends Component<NavLinksProps, {}> {
...
-        )}
+        <div className='nav-link' key='language-dropdown'>
+          <select
+            className='nav-link-lang-dropdown'
+            onChange={this.handleLanguageChange}
+            value={i18n.language as string}
+          >
+            {locales.map(lang => (
+              <option key={'lang-' + lang} value={lang}>
+                {langDisplayNames[lang]}
```
[the remark this answers is not in the document]
Comment (moT01) [reply client/src/components/Header/components/nav-links.tsx:197]: What about "change language" as the drop down without showing English
```
Change Language v
  English <check>
  Chinese
  Espanol
  ...
```

[the remark this answers is not in the document]
Comment (Nirajn2311) [reply client/src/components/Header/components/nav-links.tsx:197]: Wouldn't it be better if "Change language" was shown first and below it the current language with the chevron. Something like:
```
Change Language
English v
  Chinese
  Espanol
  ...
```

This way you'll know the current chosen language also.

File: client/src/components/Header/components/nav-links.tsx [hunk 2]
```diff
@@ -49,17 +49,36 @@ const mapDispatchToProps = {
...
+
   toggleTheme(currentTheme = 'default', toggleNightMode: any) {
     toggleNightMode(currentTheme === 'night' ? 'default' : 'night');
   }
 
+  handleLanguageChange = (
+    event: React.ChangeEvent<HTMLSelectElement>
+  ): void => {
+    this.props.toggleDisplayMenu();
+    void navigate({
```
[the remark this answers is not in the document]
Comment (shangyilim) [reply client/src/components/Header/components/nav-links.tsx]: thanks for the pointers! got it fixed

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Thanks for sorting out the navigation, the latest changes look good :+1:   We'll definitely want to double-check on staging, though @raisedadead 

I did notice one weird quirk: if you have the dropdown open on mobile, it now takes two clicks to close the menu.  The first click collapses the dropdown and the second collapses the menu:
Menu

**Comment by shangyilim** [comment]
@ojeytonwilliams, thanks for the comment. I intentionally left the menu open because the select element lives inside the nav bar. If we were to hide the menu immediately, the drop doen options will be gone as well. I'm not sure what the expected behavior should be.

**Comment by raisedadead** [comment]
I think the current behaviour of the menu is fine, @shangyilim thanks for your hard work on this.

**Review by ojeytonwilliams — DISMISSED** [review]
LGTM :+1: 

I slightly prefer having a single click collapse everything, but that's just a minor thing. This PR is solid and solves the original problem.  Thanks for putting this together and for the productive discussion @shangyilim

**Review by moT01 — COMMENTED** [review]
This is looking good 👍 It's a little tricky to test, but here's my findings and concerns...

When I open a link locally, I get redirected to `http://chinese./learn` or `http://www./italian/learn` - I am going to assume that will work in production - I guess I expected to see a `localhost` in there or something.

The styling doesn't look too far off from a browsers default style. I am fine with this for now, but I would have preferred maybe some more padding and something that looks a little more like the rest of the buttons.

This appears to open links in the same tab (with redirection). The production behavior opens links in a new tab. I'm not sure if there was a reason for that. I'm fine using the redirect. If we keep that, this will also close https://github.com/freeCodeCamp/freeCodeCamp/issues/43639

**Comment by ojeytonwilliams** [comment]
> When I open a link locally, I get redirected to http://chinese./learn or http://www./italian/learn - I am going to assume that will work in production - I guess I expected to see a localhost in there or something.

That is a bit odd, actually.  Could you copy `sample.env` over `.env` (clean cache, etc. etc.) and check it's still happening?

**Comment by ojeytonwilliams** [comment]
Thanks @moT01, it worked on GitPod, so potentially it's just a local config issue.  Still needs investigating before we merge, though.

**Review by ojeytonwilliams — APPROVED** [review]
Sorry for the delay on this, @shangyilim.  Tom's right - this just a local dev issue, not a real problem.

Thanks for your patience and hard work on this. LGTM :+1:

**Review by moT01 — APPROVED** [review]
LGTM @shangyilim 🎉 

Hey @ahmadabdolsaheb - want to take a quick look at this and make sure we're okay with the drop down style for now. There's images in comments above, you don't need to run it.

**Comment by ojeytonwilliams** [comment]
Good catch, Nick.  The select element just needs expanding to fill the parent.  Probably best to do this before merging, so we don't forget.

**Comment by ahmadabdolsaheb** [comment]
Thanks @shangyilim for the pr.  I agree with Nick and Oliver about adjusting the width of the select element in this pr. We can customize the select further in future iterations.

**Pull request merged**